comms/internal/rpcz/rate_limit_test.go

```diff
@@ -0,0 +1,147 @@
+package rpcz
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"comms.audius.co/config"
+	"comms.audius.co/db"
+	"comms.audius.co/jetstream"
+	"comms.audius.co/misc"
+	"comms.audius.co/schema"
+	"github.com/nats-io/nats-server/v2/server"
+	"github.com/nats-io/nats-server/v2/test"
+	"github.com/nats-io/nats.go"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRateLimit(t *testing.T) {
+	var err error
+
+	// reset tables under test
+	_, err = db.Conn.Exec("truncate chat cascade;")
+	assert.NoError(t, err)
+
+	// Connect to NATS and create JetStream Context
+	opts := server.Options{
+		Host:      "127.0.0.1",
+		Port:      4222,
+		JetStream: true,
+	}
+	natsServer := test.RunServer(&opts)
+	defer natsServer.Shutdown()
+	nc, err := nats.Connect(nats.DefaultURL)
+	assert.NoError(t, err)
+	defer nc.Close()
+	js, err := nc.JetStream(nats.PublishAsyncMaxPending(256))
+	assert.NoError(t, err)
+	jetstream.SetJetstreamContext(js)
+
+	// Create rate limit KV
+	SetupRateLimitRules()
+
+	// Replce rules with test rules
+	kv, err := js.KeyValue(config.RateLimitRulesBucketName)
+	assert.NoError(t, err)
+	testRules := map[string]string{
+		config.RateLimitTimeframeHours:             "24",
```

Comment:
as mentioned below limits should probably be int everywhere in our code.

Reply:
https://github.com/AudiusProject/audius-protocol/pull/4542/commits/d10ed4d852d0cbaf407939388089949b90097945